src/capture.cpp

```diff
@@ -25,111 +29,42 @@ acquisition::Capture::~Capture(){
     system_->ReleaseInstance();
 
     delete dynamicReCfgServer_;
-    
+    // for boost thread
+    if (pubThread_) {
+        pubThread_->interrupt();
```

Comment:
What are these threads doing?

Reply:
it is the boost thread for capture::run(), 

when sig int is called, usually it's propagated to boost::threads 
in destructor for obj, calling interrupt for the thread and waiting for it to join() 
// reset of pubthread releases the <boost::thread> obj

all this was there before as well in capturenodelet class